src/d3-cloud.directive.js

```diff
@@ -37,173 +37,191 @@
+            } || function () {
+              return ~~(Math.random() * 2) * 90 - 45;
+            };
+          var slopeBase = $attrs.slopeBase ? Number($scope.slopeBase) : 2;
+          var slopeFactor = $attrs.slopeFactor ? Number($scope.slopeFactor) : 30;
+
+          $scope.createCloud = function (words) {
+            var cloudWidth = $element[0].clientWidth + 0;
+            var cloudHeight = $element[0].clientHeight + 0;
+            var minScore = 0;
+            var maxScore = 1;
+            var slope = 1;
+
+            words.map(function (d) {
+              if (minScore > d.score) {
+                minScore = d.score;
+              }
+              if (maxScore < d.score) {
+                maxScore = d.score;
+              }
+            });
 
-        $scope.updateCloud = function(words) {
-          var cloudWidth = $element[0].clientWidth + 0;
-          var cloudHeight = $element[0].clientWidth + 0;
+            if (maxScore !== minScore) {
+              slope = slopeFactor / (maxScore - minScore);
+            }
 
-          var minScore = 0;
-          var maxScore = 1;
-          var slope = 1;
+            $scope.cloud = d3.layout.cloud().size([cloudWidth, cloudHeight]);
+            $scope.cloud
+              .words(words.map(function (d) {
+                const result = {
+                  text: d.name,
+                  size: d.score * slope + slopeBase
+                };
+                if (d.color) {
+                  result.color = d.color;
+                }
+                return result;
+              }))
+              .padding(padding)
+              .rotate(rotate)
+              .font($scope.font)
+              .fontSize(function (d) {
+                return d.size;
+              })
+              .on('end', draw)
+              .start();
+          };
+
+          $scope.updateCloud = function (words) {
+
+            var cloudHeight = $element[0].clientHeight;
+            var cloudWidth = $element[0].clientWidth;
+            var minScore = 0;
+            var maxScore = 1;
+            var slope = 1;
+
+            words.map(function (d) {
+              if (minScore > d.score) {
+                minScore = d.score;
+              }
+              if (maxScore < d.score) {
+                maxScore = d.score;
+              }
+            });
 
-          words.map(function(d) {
-            if (minScore > d.score) {
-              minScore = d.score;
-            }
-            if (maxScore < d.score) {
-              maxScore = d.score;
+            if (maxScore !== minScore) {
+              slope = slopeFactor / (maxScore - minScore);
             }
-          });
 
-          if (maxScore !== minScore) {
-            slope = slopeFactor / (maxScore - minScore);
+            $scope.cloud = d3.layout.cloud().size([cloudWidth, cloudHeight]);
+            $scope.cloud
+              .words(words.map(function (d) {
+                const result = {
```

Comment:
Don't use `const` here, just use `var`. Better legacy support, and result is not used as constant..